src/controllers/router-subscriber/index.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactNode } from 'react';
+
+import { useRouterStore } from '../router-store';
+import { RouterActionsType, RouterState } from '../router-store/types';
+
+export type RouterSubscriberProps = {
+  children: (state: RouterState, actions: RouterActionsType) => ReactNode;
+};
+
+export const RouterSubscriber = ({ children }: RouterSubscriberProps) => {
+  const [state, { listen, ...actions }] = useRouterStore();
+
+  if (!state.unlisten && !state.isStatic) {
```

Comment:
@albertogasparin Do you know if it's safe to remove this condition? I think it should be expected that the `Router` container is initialised by consumers.

Reply:
The truth is... I'm no longer sure how legacy pages work in this context. As we now have a router there too, but (if I recall correctly) this was meant to fix cases where we did not fully initialise. I'd recommend to keep as is (to avoid issue with the well overdue refactoring that you are doing) and maybe re-evaluate with a patch directly in Jira to see if something breaks